tests/integration/components/sl-calendar-test.js

```diff
@@ -0,0 +1,773 @@
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    this.on( 'testAction', ( dateContent ) => {
+
+        // This group of asserts verifies the multiple events on Sep. 17th
+        if ( 'Event 1 Today!' === dateContent[ 0 ].label ) {
+            assert.equal(
+                dateContent[ 0 ].date.toString(),
+                'Sat Sep 17 2022 00:00:00 GMT-0500 (CDT)',
+                'The date property was passed through'
+            );
+
+            assert.equal(
+                dateContent[ 0 ].label,
+                'Event 1 Today!',
+                'The label property was passed through'
+            );
+
+            assert.equal(
+                dateContent[ 1 ].date.toString(),
+                'Sat Sep 17 2022 00:00:00 GMT-0500 (CDT)',
+                'The date property was passed through'
+            );
+
+            assert.equal(
+                dateContent[ 1 ].label,
+                'Event 2 Today!',
+                'The label property was passed through'
+            );
+
+            assert.equal(
+                dateContent[ 2 ].date.toString(),
+                'Sat Sep 17 2022 00:00:00 GMT-0500 (CDT)',
+                'The date property was passed through'
+            );
+
+            assert.equal(
+                dateContent[ 2 ].label,
+                'Event 3 Today!',
+                'The label property was passed through'
+            );
+        // This group of asserts verifies the event on Sep. 20th
+        } else {
+            assert.equal(
+                dateContent[ 0 ].date.toString(),
+                'Tue Sep 20 2022 00:00:00 GMT-0500 (CDT)',
+                'The date property was passed through'
+            );
+
+            assert.equal(
+                dateContent[ 0 ].label,
+                'Event 1 Another Day!',
+                'The label property was passed through'
+            );
+        }
+    });
+
+    this.$( '.sl-calendar .active' ).click();
+});
+
+test( 'Setting viewMode modifies the view correctly', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            viewMode="months"
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        currentYear,
+        'The current year is set correctly'
+    );
+
+    assert.equal(
+        this.$( '.sl-calendar .active' ).text().trim(),
+        'Sep',
+        'The current month is set correctly'
+    );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
```

Comment:
What role does this serve in this test? The component will still render as expected, especially within the context of this test, without this.